### DISCUSSION TITLE
Make a suggestion

### DISCUSSION THREAD

**Discussion by xiaofanshifu** [AdguardTeam/AdGuardHome#3667] [Ideas]
Can the DNS service of adguardhome be more intelligent?

For example, according to the "top queried domains" (the first 50 records or 100 records), set ADG to automatically query the IP address corresponding to the domain name in the background at a fixed time (10 seconds, 20 seconds or 30 seconds), so that the client can directly obtain the cache data of ADG without ADG querying the corresponding IP address through the upstream DNS, which can save some time.

At the same time, with the upstream DNS server set to "fast IP address", I believe this can get a better use experience.

**Comment by ameshkov** [comment]
AGH uses a so-called "optimistic cache" in v0.107 which solves the issue: https://github.com/AdguardTeam/AdGuardHome/issues/2145

**Comment by panayotovip** [comment]
can i block services per IP ?
